packages/freezed/lib/src/freezed_generator.dart

```diff
@@ -102,12 +102,6 @@ class FreezedGenerator extends ParserGenerator<GlobalData, Data, Freezed> {
     @required bool shouldUseExtends,
   }) {
     // TODO: verify _$name is mixed-in
-    if (!element.isAbstract) {
```

Comment:
Do we want to reverse the exception? Throw if the class is abstract

Reply:
I thought about it but that would be breaking. Keeping as abstract doesn't break anything at all so we could maybe just print a warning advising people to get rid of the `abstract` ?

Reply:
A print is fine. Make sure to test it too 🙂

Reply:
I honestly have no idea how to test it, will try to look into it.

Reply:
I can't find a proper way to do so, any idea ?

Reply:
See source_gen_test.dart

You can pass expected logs in its annotation

Reply:
Yeah but the thing is it doesn't throw, just prints a warning. `source_gen_dest` doesn't seem to have api to check the logs. Is it *that* necessary to test ? It doesn't really change from the past behavior and it's just a warning ?